lib/cylc/task_events_mgr.py

```diff
@@ -681,10 +681,7 @@ def _process_message_submit_failed(self, itask, event_time):
             "time_submit_exit": get_current_time_string(),
             "submit_status": 1,
         })
-        try:
-            del itask.summary['submit_method_id']
-        except KeyError:
-            pass
+        itask.summary['submit_method_id'] = None
```

Comment:
(this was the cause of one of the failed tests - deleting job ID from the summary, on job submission failure)

Reply:
(the other was `quote(None)`)